### ISSUE TITLE
Stamps not copied when appending a page

### ISSUE THREAD

**Issue by pubpub-zz** [py-pdf/pypdf#1671]
> I recommend to use append() instead that add_page() that is pulling more data as you are copying the whole document this will be more simple:
> 
> ```
> from pypdf import PdfWriter
> 
> writer = PdfWriter()
> writer.append('sample.pdf')
> writer.write('new_sample.pdf')
> ```
> 
> Can you check you are getting what you are expecting ?

This approach works, however I still have the same result in the new file - the stamp annotation is not there.

_Originally posted by @tiago719 in https://github.com/py-pdf/pypdf/discussions/1670#discussioncomment-5126886_

**Comment by tiago719** [comment]
Just to add more context, I am trying to iterates over the pages of a PDF file, removes specified annotations, and generates a new PDF file as output.
To simplify the example presented below, I have only included code for iterating over the pages and writing them to the new PDF
Upon opening the resulting "new_sample.pdf" file with Acrobat, it can be observed that the stamp annotation that existed in the original file has been unexpectedly removed.

`from pypdf import PdfReader, PdfWriter

reader = PdfReader('sample.pdf')
writer = PdfWriter()
for page in reader.pages:
    # Some operations that will change the page object
    writer.add_page(page.get_object())

with open('new_sample.pdf', 'w+b') as f:
    writer.write(f.name)`

I tried implementing the changes suggested by @pubpub-zz:

`from pypdf import PdfWriter

writer = PdfWriter()
writer.append('sample.pdf')
writer.write('new_sample.pdf')`

But I got the same result - the stamp annotation is not in the new pdf file.

**Comment by tiago719** [comment]
Just to add more context, I am trying to iterates over the pages of a PDF file, removes specified annotations, and generates a new PDF file as output.
To simplify the example presented below, I have only included code for iterating over the pages and writing them to the new PDF
Upon opening the resulting "new_sample.pdf" file with Acrobat, it can be observed that the stamp annotation that existed in the original file has been unexpectedly removed.

```
from pypdf import PdfReader, PdfWriter

reader = PdfReader('sample.pdf')
writer = PdfWriter()
for page in reader.pages:
# Some operations that will change the page object
writer.add_page(page.get_object())

with open('new_sample.pdf', 'w+b') as f:
writer.write(f.name)
```


I tried implementing the changes suggested by @pubpub-zz:

```from pypdf import PdfWriter

writer = PdfWriter()
writer.append('sample.pdf')
writer.write('new_sample.pdf')
```

But I got the same result - the stamp annotation is not in the new pdf file.
[sample.pdf](https://github.com/py-pdf/pypdf/files/10849075/sample.pdf)

**Comment by pubpub-zz** [comment]
@tiago719,
I've understood the problem : the "image" of the stamp file is stored in the names. working on a PR

**Comment by tiago719** [comment]
Thank you very much, that is great news!

**Comment by pubpub-zz** [comment]
@tiago719 
I made a mistake in my analysis the issue was not because of Names correspondance but in some atttribute copy.
this is normally solved with #1635 
Can you confirm my tests

**Comment by tiago719** [comment]
Yap, that fixed the bug. Thanks

**Comment by MartinThoma** [comment]
Closed by #1635
